packages/grafana-ui/src/components/VizLegend/SeriesIcon.tsx

```diff
@@ -29,28 +31,47 @@ export const SeriesIcon = React.memo(
       cssColor = color!;
     }
 
-    const styles: CSSProperties = {
-      background: cssColor,
-      width: '14px',
-      height: '4px',
-      borderRadius: theme.shape.radius.pill,
-      display: 'inline-block',
-      marginRight: '8px',
-    };
+    const lineStyleStyle =
+      lineStyle?.fill === 'dot' ? styles.dot : lineStyle?.fill === 'dash' ? styles.dash : styles.solid;
+
+    const colorStyle: CSSProperties =
+      lineStyle?.fill === 'dot' || lineStyle?.fill === 'dash'
+        ? {
+            borderTopColor: cssColor,
+          }
+        : {
+            backgroundColor: cssColor,
+          };
 
     return (
       <div
         data-testid="series-icon"
         ref={ref}
-        className={cx(className, styles2.forcedColors)}
-        style={styles}
+        className={cx(className, styles.forcedColors, styles.container, lineStyleStyle)}
+        style={colorStyle}
         {...restProps}
       />
     );
   })
 );
 
-const getStyles = () => ({
+const getStyles = (theme: GrafanaTheme2) => ({
+  container: css({
+    marginRight: '8px',
+    display: 'inline-block',
+    width: '14px',
+  }),
+  dot: css({
+    borderTop: 'dotted 3.99px',
```

Comment:
3.99, huh? :sweat_smile:

Reply:
it feels a lot more affordable than 4 :D 

using `4` makes only 2 dots show up with a large gap *on my machine*, hence the odd "hack"